### PR TITLE
fix(UI): change the button style in searchbar

### PR DESCRIPTION
**Pull request by mmatsumoto1026**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

  Description of changes is edited at 9/20 since the height of navbar and searchbar was changed in another PR, therefore appearance of the search and reset button when the camper hovers over them in the latest code was also changed.

This PR was opened to change the background color, ~~size,~~ and the svg `fill` color of the search and reset button in the search bar when the camper hovers over them so that solving the issues described bellow.

Issues:
- When the night mode is on, the contrast between the background color of the buttons when the camper hovers over them(`--gray-05` #f5f6f7) and the `fill` color of `Magnifier` and `InputReset` svg image(`--gray-15` #d0d0d5) is low([contrast ratio](https://webaim.org/resources/contrastchecker/) is 1.42:1) so the svg icon images might not be easily recognized by people with moderately low vision as shown in bellow image.
  <details>
    <summary>Image: Low contrast ratio between background color of the buttons and svg icon image</summary>

    nightmode
  </details>

- ~~Again When the night mode is on, the background color of the buttons when the camper hovers over them(`--gray-05` #f5f6f7) is quite light compared to the background color of universal nav bar(`--theme-color` #0a0a23), therefore it looks inflated and pops out a bit by [contrast effect with the background or neighboring colors](https://www.hy-factory.net/read/expansion_shrink_color_en.php). (This may differ from person to person)~~
- ~~When the night mode is off, the contrast between the background color of the buttons when the camper hovers over them(`--gray-85` #1b1b32) and the background color of universal nav bar(`--theme-color` #0a0a23) is low, so when the camper hovers over the reset button, it looks like the "X" reset button  image is popped out from the search form input as shown in bellow image.~~

  [9/20 image deleted]

Solutions suggested in this PR(result is as shown in bellow image):
- Change the background color of the buttons and the fill color of the SVG image so that the contrast when the camper hovers over them is high enough(contrast ratio is 5.98:1).
- ~~Resize the buttons to make them a bit smaller to avoid having they look like inflated and pops out a bit when the camper hovers over them, and also making them look more like a button.~~
- Unify the background color of the button when night mode is on and off, ~~to solve the issue when night mode is off.~~ so the contrast ratio will be always constant.

  [9/20 image deleted]

I've tested the code in Chrome, Firefox, Edge, and Brave on windows 11 PC locally, and also Chrome, Safari, Opera, and Brave on iPhone SE using the ngrok.

Test condition:
- All elements in search bar are fully responsive.
- From 80% to 400% in zoom, no collapsion or overlaping of elements are occured.

### CODE REVIEW

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
I don't believe we should be rounding the corners. That doesn't match the aesthetic of any of the other buttons on our platform.

**Comment by mmatsumoto1026** [comment]
Thanks for your review, @naomi-lgbt.

And I apologize for suggesting a style for the buttons that doesn't fit with the aesthetic of any of the other buttons on fCC's platform. That was simply my overlooking, and as a result of researching various search bars when drafting this pull request. So I agree with you and would push the code without `border-radius` property after testing locally.

**Comment by ahmaxed** [comment]
Thank you for your PR. I see a small space between the button and its container that did not exist previously. If that is adjusted, this pr should be good to go.

**Comment by mmatsumoto1026** [comment]
Thanks for your review, @ahmaxed.

The height of the buttons were adjusted to match the height of the search box.

File: client/src/components/search/searchBar/searchbar.css [hunk 1]
```diff
@@ -19,33 +19,60 @@
...
   display: inline-block;
-  height: 26px;
+  height: var(--header-element-size);
 }
 
 .ais-SearchBox-submit,
 .ais-SearchBox-reset {
-  height: 26px;
+  height: var(--header-element-size);
   width: 26px;
```
Comment (ahmaxed) [inline client/src/components/search/searchBar/searchbar.css:52]:
```suggestion
  width: var(--header-element-size);
```

**Review by ahmaxed — APPROVED** [review]

File: client/src/components/search/searchBar/searchbar.css [hunk 2]
```diff
@@ -19,33 +19,60 @@
...
-  height: 26px;
+  height: var(--header-element-size);
 }
 
 .ais-SearchBox-submit,
 .ais-SearchBox-reset {
-  height: 26px;
-  width: 26px;
+  height: var(--header-element-size);
+  width: var(--header-element-size);
```
Comment (Sboonny) [inline client/src/components/search/searchBar/searchbar.css:52]:
```suggestion
  width: 26px;
```

**Review by moT01 — APPROVED** [review]
LGTM :tada:

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**